src/gov-action-contracts/nonemergency/AddNovaKeysetAction.sol

```diff
@@ -0,0 +1,28 @@
+
+    function perform() public {
+        // adds new keyset as valid submitter
+        novaInbox.setValidKeyset(keysetBytes);
+
+        // verify keyset was set correctly
+        require(novaInbox.isValidKeysetHash(keysetHash), "valid keyset not set");
+
+        // this action does not remove the old keyset to allow for a seemless transition on the batch poster
+        // once the offchain batch poster is updated, the old keyset should be removed so the one added in this action is the sole valid one
```

Comment:
Does this mean we'll have another action contract in future to call `invalidateKeysetHash`?

Reply:
yes